### PR TITLE
Fixes to make sanitizers happy

### PR DESCRIPTION
**Pull request by vlovich**
Address TSAN discussion from https://groups.google.com/g/capnproto/c/634juhn5ap0/m/pyRiwWl1AAAJ
Address UBSAN discussion in #1079 per (at least as I understood) @kentonv's suggestion.

### CODE REVIEW

File: c++/src/capnp/layout.c++ [hunk 1]
```diff
@@ -30,26 +30,47 @@
...
+#endif
+#endif
+
+
 namespace capnp {
 namespace _ {  // private
 
 #if !CAPNP_LITE
-static BrokenCapFactory* brokenCapFactory = nullptr;
+static BrokenCapFactory* gBrokenCapFactory = nullptr;
```
Comment (kentonv) [inline c++/src/capnp/layout.c++:48]: This `g` prefix isn't idiomatic KJ, but I'd accept `globalBrokenCapFactory`.

File: c++/src/capnp/layout.c++ [hunk 2]
```diff
@@ -30,26 +30,47 @@
...
 #elif _MSC_VER
-  *static_cast<BrokenCapFactory* volatile*>(&brokenCapFactory) = &factory;
+  *static_cast<BrokenCapFactory* volatile*>(&gBrokenCapFactory) = &factory;
 #else
 #error "Platform not supported"
 #endif
 }
 
+static BrokenCapFactory* readGlobalBrokenCapFactoryForLayoutCpp() {
+#if IS_BUILDING_WITH_TSAN()
```
Comment (kentonv) [inline c++/src/capnp/layout.c++:65]: Let's just have it always be atomic and skip the TSAN check. No need to comment about TSAN either -- it's reasonable enough that if the global is atomically written, it ought to be atomically read.

File: c++/src/capnp/layout.c++ (hunk 1 above)
Comment (vlovich) [reply c++/src/capnp/layout.c++]: Sorry about that. Switching between different codebases.  Definitely prefer KJ's style.

File: c++/src/capnp/layout.c++ (hunk 2 above)
Comment (vlovich) [reply c++/src/capnp/layout.c++]: Good call.

**Comment by kentonv** [comment]
Thanks.

**Pull request merged**